protecode/scanning.py

```diff
@@ -478,6 +486,15 @@ def process(
                     product_id=scan_result.product_id(),
                 )
 
+            if delete_inactive_products_after and scan_request.target_product_id:
+                # remove deletion flag for current product as it is still in use
+                self.protecode_client.set_metadata(
+                    product_id=scan_request.target_product_id,
+                    custom_attribs={
+                        'DELETION_AFTER': '', # must be empty str and not `None`
```

Comment:
+1 for adding a comment indicating empty str should remain.

however, two remarks:
- we might adjust `set_metadata` in such a way that the function will translate `None` -> `''` . that way, our API might maybe feel a bit more "natural" (not sure whether you agree)
- it would be nice to explain _why_ we need empty str (I assume to prevent `None` (or `null`?) (str literal) to be stored, which would cause parsing-issues (and also not semantically be what we want - which is to remove the attribute)